server/channels/app/post_metadata.go

```diff
@@ -190,16 +190,19 @@ func (a *App) getEmbedsAndImages(c request.CTX, post *model.Post, isNewPost bool
 	return post
 }
 
-func (a *App) sanitizePostMetadataForUserAndChannel(c request.CTX, post *model.Post, previewedPost *model.PreviewPost, previewedChannel *model.Channel, userID string) *model.Post {
-	if post.Metadata == nil || len(post.Metadata.Embeds) == 0 || previewedPost == nil {
-		return post
+// shouldSanitizePostMetadataForUserAndChannel returns true/false whether the user
+// has permission to view the metadata, and thereby the metadata should be
+// wiped off or not.
+func (a *App) shouldSanitizePostMetadataForUserAndChannel(c request.CTX, post *model.Post, previewedChannel *model.Channel, userID string) bool {
```

Comment:
FWIW, I'm still in favour of inlining this. It's far too specific to the permalink preview case to be useful in any other context, and it would actually eliminate more code since some of these checks are redundant.

Reply:
Yeah, the thing is then this becomes hard to unit test. Let me see.